packages/hoppscotch-common/src/helpers/curl/curlparser.ts

```diff
@@ -28,12 +28,31 @@ import {
 
 const defaultRESTReq = getDefaultRESTRequest()
 
+const parseCurlArguments = (curlCommand: string) => {
+  let args: parser.Arguments = parser(curlCommand)
+
+  if (
+    objHasProperty("dataUrlencode", "string")(args) ||
+    objHasProperty("dataUrlencode", "object")(args)
+  ) {
+    const urlEncodedData: string[] = Array.isArray(args.dataUrlencode)
+      ? args.dataUrlencode
+      : [args.dataUrlencode]
+
+    const data = A.map((key: string) => decodeURI(key))(urlEncodedData)
+
+    args = { ...args, d: data }
+  }
+
+  return args
+}
+
```

Comment:
Can you refactor your code using `fp-ts` since the entire parser is built with it? Despite that, your solution works well.

Reply:
@anwarulislam 
I'm just learning how to write code in a declarative format. I tried to write the code, but I couldn't. I will be glad if you can correct me.

```
pipe(
      parser(curlCommand),
      O.fromPredicate(
        (args: parser.Arguments) =>
          objHasProperty("dataUrlencode", "string")(args) ||
          objHasProperty("dataUrlencode", "object")(args)
      ),
      O.map((args: parser.Arguments) =>
        flow(
          (args: parser.Arguments) =>
            Array.isArray(args.dataUrlencode)
              ? args.dataUrlencode
              : [args.dataUrlencode],
          A.map((key: string) => decodeURI(key)),
          O.map((data) => ({ ...args, d: data }))
        )
      ),
      O.altW((data) => console.log(data))
    )
```